pkg/resource/deploy/providers/registry.go

```diff
@@ -304,14 +294,9 @@ func (r *Registry) Diff(urn resource.URN, id resource.ID, olds, news resource.Pr
 	}
 
 	// If the diff requires replacement, unload the provider: the engine will reload it during its replacememnt Check.
-	//
-	// If the diff does not require replacement but does have changes, and we are running a preview, register it under
-	// its current ID so that references to the provider from other resources will resolve properly.
 	if diff.Replace() {
 		closeErr := r.host.CloseProvider(provider)
 		contract.IgnoreError(closeErr)
-	} else if r.isPreview && diff.Changes == plugin.DiffSome {
-		r.setProvider(mustNewReference(urn, id), provider)
```

Comment:
> What impact do the changes here have in practice on when providers are configured? I recall this was added originally to address a significant performance issue around configuring providers multiple times. But perhaps the removal of both this and the section above address that?

These changes should only push provider configuration later in the program, and only during previews. Instead of being configured in `Check` and `Diff` when previewing, the provider will be configured when the `Create` or `Update` step is performed. This should result in the same number of calls to `Configure`.